src/pages/SubmissionConfig.test.js

```diff
@@ -1307,6 +1307,150 @@ describe('SubmissionConfig inputs/outputs definitions', () => {
     within(cells3[4]).getByText('bar_string')
   })
 
+  it('should populate fields from data table on click', async () => {
+    const mockRunSetResponse = jest.fn(() => Promise.resolve(runSetResponseSameInputNames))
+    const mockMethodsResponse = jest.fn(() => Promise.resolve(methodsResponse))
+    const mockSearchResponse = jest.fn((_, recordType) => Promise.resolve(searchResponses[recordType]))
+    const mockTypesResponse = jest.fn(() => Promise.resolve(typesResponse))
+
+    await Ajax.mockImplementation(() => {
+      return {
+        Cbas: {
+          runSets: {
+            getForMethod: mockRunSetResponse
+          },
+          methods: {
+            getById: mockMethodsResponse
+          }
+        },
+        Wds: {
+          search: {
+            post: mockSearchResponse
+          },
+          types: {
+            get: mockTypesResponse
+          }
+        }
+      }
+    })
+
+    render(h(SubmissionConfig))
+
+    await waitFor(() => {
+      expect(mockRunSetResponse).toHaveBeenCalledTimes(1)
+      expect(mockTypesResponse).toHaveBeenCalledTimes(1)
+      expect(mockMethodsResponse).toHaveBeenCalledTimes(1)
+      expect(mockSearchResponse).toHaveBeenCalledTimes(1)
+    })
+
+    const button = await screen.findByRole('button', { name: 'Inputs' })
+    await fireEvent.click(button)
+
+    const table = await screen.findByRole('table')
+    const rows = within(table).queryAllByRole('row')
+    const cells1 = within(rows[1]).queryAllByRole('cell')
+    const cells2 = within(rows[2]).queryAllByRole('cell')
+    const cells3 = within(rows[3]).queryAllByRole('cell')
+
+    within(cells1[0]).getByText('foo')
+    within(cells1[1]).getByText('foo_rating')
+    within(cells1[2]).getByText('Int')
+    within(cells1[3]).getByText('None')
+    within(cells1[4]).getByText(/Use /)
+    const inputFillButton = within(cells1[4]).getByText('foo_rating')
+    within(cells1[4]).getByText(/ from data table?/)
```

Comment:
I was wondering about this - what does it look like when a required attribute is present in the data table? Would you mind taking a screenshot?

Reply:
<img width="1425" alt="Screenshot 2023-06-02 at 11 09 15 AM" src="https://github.com/DataBiosphere/cbas-ui/assets/67511512/1530d064-9c96-4166-95f3-6a6d17494637">

Same text content, not italic since not optional, and maintains the ⚠️ This input is required warning. I can change this to a required input in the tests to make sure it still works.